src/components/LocationAutocomplete/LocationAutocomplete.tsx

```diff
@@ -11,12 +10,47 @@ import POWERED_BY_GOOGLE_ON_WHITE_2X from './images/powered_by_google_on_white@2
 import POWERED_BY_GOOGLE_ON_WHITE_3X from './images/powered_by_google_on_white@3x.png';
 import styles from './LocationAutocomplete.scss';
 
+interface Props {
+    /** input field ref */
+    inputRef?: React.RefObject<HTMLInputElement>;
+    /** to be shown in the input field when no value is typed */
+    inputPlaceholder: string;
+    /** default input value */
+    defaultInputValue?: string;
+    /** to be shown when no suggestions are available */
+    noSuggestionsPlaceholder: string;
+    /** trigger of the initial focus of the input field */
+    isFocused?: boolean;
+    /** label for the Clear button */
+    clearLabel?: string;
+    /** defines if there's a single location to select in component */
+    singleLocation?: boolean;
+    /** callback to be called with selected value. */
+    onSelectionChange: (value: google.maps.places.AutocompletePrediction) => void;
+    /** restrict predictions to country/countries.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#ComponentRestrictions
+     */
+    country?: string | string[];
+    /** type of locations that should be searched for.
+     * For details see: https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletionRequest.types
+     */
+    placeTypes?: string[];
+    /** show state and country in suggestions list */
+    showCountryInSuggestions?: boolean;
+    /** function to remove all locations */
+    onRemoveAllLocations?: () => void;
+    /** function to be executed if error occurs while fetching suggestions */
+    onError?: (error: google.maps.places.PlacesServiceStatus) => void;
+    /** To hide powered by google logo. For legal reasons only set it to true if Google map is displayed on the same screen as this component! */
+    hidePoweredByGoogleLogo?: boolean;
+}
+
 const { elem } = bem('LocationAutocomplete', styles);
 
 const DEBOUNCE_DELAY = 350;
 const ACCEPTABLE_API_STATUSES = ['OK', 'NOT_FOUND', 'ZERO_RESULTS'];
```

Comment:
Are these candidates for `src/constants/`?

Reply:
perhaps. but I didn't want to make any other changes then what is needed for TS migration. There are a few things "funny" with these components, like it uses `function` definitions instead of `const`. But If I change all that, the PR will be very difficult to review, and more changes for bugs.